Source/ACE/Entity/Player.cs

```diff
@@ -45,6 +45,11 @@ public sealed class Player : MutableWorldObject
         public bool IsOnline { get; private set; }
 
         /// <summary>
+        /// ObjectId of the currently selected Target (only players and creatures)
+        /// </summary>
+        public uint SelectedTarget { get; set; }

```

Comment:
I am not sure if you are right: if you target a creature or player a QueryHealth message is fired with the objectID of the targeted creature or player. If you target an item like a lifestone a QueryHealth message is fired too, but with an objectID of 0. The same happens if you press the ESC key to deselect your target,